### PR TITLE
[product] Pricelist report: Convert to New Api and Make group by category ..

### PR DESCRIPTION
**Pull request by atp-odoo**
Task : https://www.odoo.com/web?#id=9967&view_type=form&model=project.task&menu_id=3942&action=327
pad : https://pad.odoo.com/p/r.xH2I0dV8w1thCBt1

### CODE REVIEW

**Comment by atp-odoo** [comment]
Follow PEP8 coding guidlines in all your code.

File: addons/product/report/product_pricelist.py [hunk 1]
```diff
@@ -2,101 +2,106 @@
...
-    def _get_pricelist(self, pricelist_id):
-        pricelist = self.pool.get('product.pricelist').read(self.cr, self.uid, [pricelist_id], ['name'], context=self.localcontext)[0]
-        return pricelist['name']
-
-    def _get_currency(self, pricelist_id):
-        pricelist = self.pool.get('product.pricelist').read(self.cr, self.uid, [pricelist_id], ['currency_id'], context=self.localcontext)[0]
-        return pricelist['currency_id'][1]
+    @api.model
+    def _get_price(self, product_ids):
+        products = []
```
Comment (hmo-odoo) [inline addons/product/report/product_pricelist.py]: need to improve code.
-> variable name
-> logic

File: addons/product/report/product_pricelist.py [hunk 2]
```diff
@@ -2,101 +2,106 @@
...
-        else:
-            res = self.pool.get('product.product').read(self.cr, self.uid, [product_id])
-            price =  self.formatLang(res[0]['list_price'], digits=sale_price_digits, currency_obj=pricelist.currency_id)
-        return price
-
+    @api.multi
+    def render_html(self, data):
+        form = data.get('form')
+        report_obj = self.env['report']
+        self.quantity = []
```
Comment (hmo-odoo) [inline addons/product/report/product_pricelist.py]: why need to assign into self ?

File: addons/product/report/product_pricelist.py [hunk 3]
```diff
@@ -2,101 +2,78 @@
...
+        self.model = self.env.context.get('active_model')
+        self._set_quantity(form)
+        self.pricelist = self.env['product.pricelist'].browse(form['price_list'])
+        selected_records = self.env[self.model].browse(data.get('ids'))
+        docargs = {
+            'doc_ids': data.get('ids'),
+            'doc_model': self.model,
+            'docs': selected_records,
+            'time': time,
+            'get_pricelist': self.pricelist.name,
```
Comment (hmo-odoo) [inline addons/product/report/product_pricelist.py]: do not add prefix 'get_'. it's value not a method. so just pricelist, currency, titles, categories enough :)

Comment (dka-odoo) [inline addons/product/report/product_pricelist.py]: :+1:

**Pull request closed without merging**